arch/riscv/chunkset_rvv.c

```diff
@@ -0,0 +1,98 @@
+/* chunkset_rvv.c - Genernal version of chunkset
+ * Copyright (C) 2023 SiFive, Inc. All rights reserved.
+ * Contributed by Alex Chiang <alex.chiang@sifive.com>
+ * For conditions of distribution and use, see copyright notice in zlib.h
+ */
+
+#include "zbuild.h"
+
+/*
+ * It's not a optimized implemantation using RISC-V RVV, but a general optimized one.
```

Comment:
I am wondering if this isn't better to be put into arch/generic and all references to RISC-V removed since there are no RISC-V intrinsics being used.

Reply:
It's very interesting problem. I'll tri it on x86/ARM when I have free time.